module-system/module-implementations/sov-chain-state/src/query.rs

```diff
@@ -0,0 +1,42 @@
+use sov_rollup_interface::zk::ValidityCondition;
+use sov_state::WorkingSet;
+
+use super::ChainState;
+use crate::{StateTransitionId, TransitionInProgress};
+
+#[derive(serde::Serialize, serde::Deserialize, Debug, Eq, PartialEq)]
+/// Structure returned by the query methods.
+pub struct Response {
+    /// Value returned by the queries
+    pub value: u64,
+}
+
+impl<C: sov_modules_api::Context, Cond: ValidityCondition> ChainState<C, Cond> {
+    /// Get the height of the current slot
+    pub fn slot_height(&self, working_set: &mut WorkingSet<C::Storage>) -> Option<u64> {
+        self.slot_height.get(working_set)
```

Comment:
Prior to this block, this get is considered infallible. Is there a reason to not treat is as such here?